objectivec/Tests/CocoaPods/run_tests.sh

```diff
@@ -104,7 +104,8 @@ cleanup() {
 }
 
 do_test() {
-  local TEST_NAME="$1"
+  local TEST_NAME=$(echo "$1" | cut -f1 -d_)
```

Comment:
How about just checking if the test has a prefix of "iOS" and then adding in the SDK args as needed?  You can build up the command to invoke as a bash array, and if it has the prefix, add the extra args (otherwise don't).